### ISSUE TITLE
0.9.3 -> 1.0.0: Cluster keeps updating with no devices changed

### ISSUE THREAD

**Issue by dimm0** [rook/rook#3113]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Update went well, but not operator keeps updating the cluster.
From logs:
```
2019-05-03 20:44:27.915700 I | op-cluster: Done creating rook instance in namespace rook
2019-05-03 20:44:27.916554 I | op-cluster: device lists are equal. skipping orchestration
2019-05-03 20:44:27.916873 I | op-cluster: Running orchestration for namespace rook after device change
```

**Expected behavior:**
Operator doesn't do updates if nothing changed

**Environment**:
* Rook version (use `rook version` inside of a Rook Pod): v1.0.0
* Kubernetes version (use `kubectl version`): 1.13.5
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm

**Comment by leseb** [comment]
@noahdesu haven't we solved that in 1.0.2?

**Comment by noahdesu** [comment]
yep. i found 3 duplicates that got closed when we merged the fixes, but this one slipped through.
